pkg/dataobj/internal/dataset/dataset_sort.go

```diff
@@ -0,0 +1,173 @@
+package dataset
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	"github.com/grafana/loki/v3/pkg/dataobj/internal/result"
+)
+
+// Sort returns a new Dataset with rows sorted by the given sortBy columns in
+// ascending order. The order of columns in the new Dataset will match the
+// order in set. pageSizeHint specifies the page size to target for newly
+// created pages.
+//
+// If sortBy is empty or if the columns in sortBy contain no rows, Sort returns
+// set.
+func Sort(ctx context.Context, set Dataset, sortBy []Column, pageSizeHint int) (Dataset, error) {
```

Comment:
When do we need to sort the data ?

Reply:
Sorting data will happen before objects are flushed. We'll want to at least sort log data by stream ID (if we're using a megatable) and timestamp. 

dataset.Sort isn't needed at all if we flush for every call to append (since we could just sort the incoming logs), but we would need it if flushes happen every few appends.